pki/tls/sni.go

```diff
@@ -32,8 +32,9 @@ func AuthoritySNITLSGetter(authority pki.Authority, logger Logger) func(*tls.Cli
 			leaf, err := authority.LeafForGroup(pki.ControllerIPLeafGroup)
 			if err != nil && !errors.IsNotFound(err) {
 				return nil, errors.Annotate(err, "fetching ip address certificate")
+			} else if !errors.IsNotFound(err) {
```

Comment:
this would be better as `} else if err == nil {`

or even the whole block as

```
if err == nil {
    cert = leaf.TLSCertificate()
} else if !errors.IsNotFound(err) {
    return nil, errors.Annotate(err, "fetching ip address certificate")
}
```